### PR TITLE
AI can view Mindshields and Wanted status.

### PR DESCRIPTION
**Pull request by ScarKy0**
## About the PR

Replaces the AI Job Icon view with a Status Icon view, which allows it to see mindshields and wanted status.

## Why / Balance

Currently just seeing job icons is underwhelming and pretty useless.

## Technical details

Renamed the ActionShowJobIcons to be Status icons instead.
Added new components to the toggle that AI is using.

## Media

image

## Requirements

- [X] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

Changed the ID of ActionShowJobIcons to ActionShowStatusIcons

**Changelog**

:cl:
- tweak: AI can now see mindshields and wanted status.

### CODE REVIEW

**Comment by DogZeroX** [comment]
In 13, ai can only see jobs and structure health borgs/mechs

**Comment by ScarKy0** [comment]
> In 13, ai can only see jobs and structure health borgs/mechs

That can be easily adjusted if needed

**Comment by lzk228** [comment]
again, same with aghost, really would be better to have toggleable menu for this

**Comment by ScarKy0** [comment]
> again, same with aghost, really would be better to have toggleable menu for this

This is a toggle
image

**Comment by lzk228** [comment]
i mean, so you can toggle which of icons you want to see now

**Comment by slarticodefast** [comment]
Please explain in the technical details section why you are removing the Clear() call in the systems. It probably is there for a reason, so make sure this does not break anything else.

**Comment by ScarKy0** [comment]
> Please explain in the technical details section why you are removing the Clear() call in the systems. It probably is there for a reason, so make sure this does not break anything else.

Thanks for reminding me, and as far as i tested it works fine, leaves no duplicate comps or anything, same with other huds

**Comment by ScarKy0** [comment]
Removed the health stuff because its bugged also kinda makes little sense.

**Comment by Plykiya** [comment]
Borgs and AI shouldn't be concerned with mindshield status or wanted status, it was already removed from borgs somewhat recently

**Comment by Sadopeer** [comment]
> Borgs and AI shouldn't be concerned with mindshield status or wanted status, it was already removed from borgs somewhat recently

Absolutely agree, ill cause more RP issues

**Comment by ScarKy0** [comment]
I think AI has more of a concern with wanted stuff
Though in that case the toggle might as well be removed so AI can always see job icons

**Comment by Plykiya** [comment]
> I think AI has more of a concern with wanted stuff
> Though in that case the toggle might as well be removed so AI can always see job icons

The AI should always, and only, be able to see job icons, just like borgs. It is the sole determination of whether someone is crew under default lawsets.

Being wanted or not having a mindshield doesn't change that.

**Comment by Emisse** [comment]
i dont want ai being sec sided, sec ai is worst ai

**Comment by superjj18** [comment]
> i dont want ai being sec sided, sec ai is worst ai

There should be some rule clarification when it comes to ai.  I've heard people refer to ai as "valid-hunting" via bolting/snitching on antags.  Is it against the rules for AI to go out of its way to aid security/command?  I would think that that's the intended ai playstyle for most of the "good" lawsets(NT default, crewsimov, robocop, paladin, etc).

I get why Borgs don't get the wanted status because they are 99.9% of the time non-combatants who shouldn't be valid hunting, but AI is supposed to be... intelligent and be constantly updated with as much information as the station gives.

**Comment by Catdere** [comment]
> i dont want ai being sec sided, sec ai is worst ai

As someone who mains AI in 13 (now 14) we tend to lean on sec most of the time they are really easy to assist and need to get into departments most of the time also if you see someone getting killed and then looted its kinda in your best interest to bolt the door. Now does this mean were vaild hunting?? no we are the station in a way a murderer being bolted off and alerted to security is fine. But if the ai player ONLY interacts with sec it kinda ruins ai as job you have so power to assist people in their jobs and thus should use that power. Even SEEING a wanted symbol would get you sidetracked and focused on helping sec detain this person, rather then seeing this person do bad and helping sec in the moment or being requested by sec to help.

**Comment by Catdere** [comment]
> > i dont want ai being sec sided, sec ai is worst ai
> 
> There should be some rule clarification when it comes to ai. I've heard people refer to ai as "valid-hunting" via bolting/snitching on antags. Is it against the rules for AI to go out of its way to aid security/command? I would think that that's the intended ai playstyle for most of the "good" lawsets(NT default, crewsimov, robocop, paladin, etc).
> 
> I get why Borgs don't get the wanted status because they are 99.9% of the time non-combatants who shouldn't be valid hunting, but AI is supposed to be... intelligent and be constantly updated with as much information as the station gives.

I think Ai snitching and bolting doors is completely ok it doesn't cause crew harm and leads to less to harm, Also think about this again but cut out solo antags... yeah the ai would kinda WANT and NEED to tell somebody about Ninjas, Nukies, Dragons, Rat Kings, Revs (The Ghostly Kind). 

As Ai you are the station, And threats to the crew and the station are threats to YOU. How i preserve this under crewsimov is helping sec falls under law 1 and law 3, and just so were clear why it would be bad if there was a rule like this heres a example 

*You see something whos died in a closed off room there's someone standing in a corner healing themselves and its clear to you the Ai that they are murderer. You are now faced with a choice you either do nothing and let the traitor get away and kill more people causing harm to crew, or you call a medical doctor to recover the body risking that crew members life and breaking law 1*

**Comment by Sarahon** [comment]
> > i dont want ai being sec sided, sec ai is worst ai
> 
> There should be some rule clarification when it comes to ai. I've heard people refer to ai as "valid-hunting" via bolting/snitching on antags. Is it against the rules for AI to go out of its way to aid security/command? I would think that that's the intended ai playstyle for most of the "good" lawsets(NT default, crewsimov, robocop, paladin, etc).
> 
> I get why Borgs don't get the wanted status because they are 99.9% of the time non-combatants who shouldn't be valid hunting, but AI is supposed to be... intelligent and be constantly updated with as much information as the station gives.

Thus far, and I've complained about it too. AI becomes "sec bot" and starts ordering sec around, which is causing a headache for HOS, Warden or Cap to organise people. I don't care if they can see jobs but AI-sec right now is just annoying to me.

**Comment by Catdere** [comment]
> > > i dont want ai being sec sided, sec ai is worst ai
> > 
> > 
> > There should be some rule clarification when it comes to ai. I've heard people refer to ai as "valid-hunting" via bolting/snitching on antags. Is it against the rules for AI to go out of its way to aid security/command? I would think that that's the intended ai playstyle for most of the "good" lawsets(NT default, crewsimov, robocop, paladin, etc).
> > I get why Borgs don't get the wanted status because they are 99.9% of the time non-combatants who shouldn't be valid hunting, but AI is supposed to be... intelligent and be constantly updated with as much information as the station gives.
> 
> Thus far, and I've complained about it too. AI becomes "sec bot" and starts ordering sec around, which is causing a headache for HOS, Warden or Cap to organise people. I don't care if they can see jobs but AI-sec right now is just annoying to me.
> 
> Giving information and doing shit is fine, but players themselves are a headache at times and can really shit up the radio.

I feel like youre misunderstanding ordering people around and being told about a threat. most of the time sec head to the location of an active threat to suppress it believe it or not the literal camera with a speaker can give pin pointing information with no risk what so ever. 

And if youre issue is the Ai says "Doctor Pomsom is breaking into caps room from maints." then you should also have a problem with "Hey medical there's a dead engi by solars can you go pick him up."

**Comment by LordEclipse** [comment]
I don't think this is good for the game in any way.
It's the same reason that cameras aren't mapped into maints roundstart. A.I. is the ultimate information gathering role, and giving them access to more tools and information without any additional effort would just cause massive balance issues. A.I. having access to all comms, but also able to see mindshields and criminal status are IMO too much. A.I. seeing mindshields would prevent people from being able to disguise as command, and if A.I. can see somebody being wanted they can find them without much additional effort, thus completely shutting down their escape. However A.I. CAN choose to read the criminal records console, as well as has access to security channel. I just don't think personally that the INFORMATION role should be given more tools to make their already easy ability to gather information, even easier. Not to mention think about malf/comp AI? This just makes command even bigger targets than they already are in all circumstances, not even mentioning the effect it has on crew. ALSO what about when Ninjas hack criminal records to turn all crew into WANTED?

**Comment by EthanQix** [comment]
HoP and Captain were given administrative HUDs, that display only job icons, because they don't need Wanted and Mindshield status to do their job. This is the same thing for AI, except even more important. AI is supposed to control access to places, opening doors, etc, so job info is important.

Giving AI vision of Wanted status, in particular, would make hiding as an antag completely impossible. Giving them Mindshield vision would do the same for revs.

Long story short : you don't want to give Big Brother any more information than absolutely necessary for their job.

**Comment by Catdere** [comment]
> s someone who mains AI in 13 (now 14) we tend to lean on sec most of the time they are really easy to assist and need to get into departments most of the time also if you see someone getting killed and then looted its kinda in your best interest to bolt the

You seem to misunderstand the Ai's job completely. The Ai is meant to be the head of Borgs ordering them to do things whilst assisting every department in many ways. 

Such as pointing out the locations of anoms to sci, Pointing out station threats to sec, Alerting parameds and medical that death has happened ect. Opening doors is like the least thing you do as Ai. 

Giving the AI the power to see wanted status ruins game balance for crooks if the Ai can just see a wanted person at a glance instead of already helping sec locate said person.

**Comment by slarticodefast** [comment]
Hey, I brought this up for maintainer discussion and the decision was to close this PR for the following reasons:
- While the AI can see the same information via the criminal records console it is not its primary job to do so
- AI can already see sec channel and hear callouts so it can protect the station against more immediate threats, but you won't need a wanted icon when someone is running down the hallway with a C20-r
- Showing the mindshield status makes is very hard for someone to disguise as sec or command

**Comment by nymhole** [comment]
wanted status is necessary,
